AudioLoader/Music.py

```diff
@@ -694,6 +699,145 @@ def available_groups(self, group):
+                    # add track to list of tracks
+                    tracks.append(track_folder)
+
+                    meta = torchaudio.info(os.path.join(
+                        track_folder, 'mixture.wav'))
+                    assert meta.sample_rate == sr
+                    track_lengths.append(meta.num_frames)
+
+        return tracks, track_lengths
+
+    def __len__(self):
+        return self._size
+
+    def _get_random_track_idx(self):
+        return random.randrange(len(self.tracks))
 
+    def _get_random_start(self, length):
+        return random.randrange(length - self.segment + 1)
+
+    def _get_track_from_chunk(self, index):
+        track_idx = np.digitize(index, self.cum_chunks)
+        if track_idx > 0:
+            chunk_start = (index - self.cum_chunks[track_idx]) * self.segment
+        else:
+            chunk_start = index * self.segment
+        return self.tracks[track_idx], chunk_start
+
+    def __getitem__(self, index):
+        stems = []
+        if self.seq_duration <= 0:
+            folder_name = self.tracks[index]
+            x, _ = torchaudio.load(
+                os.path.join(folder_name, 'mixture.wav')
+            )
+            for s in self.sources:
+                source_name = os.path.join(folder_name, s + '.wav')
+                audio, _ = torchaudio.load(source_name)
+                stems.append(audio)
+        else:
+            if self.random:
+                track_idx = index // self.samples_per_track
+                folder_name, chunk_start = self.tracks[track_idx], self._get_random_start(
+                    self.track_lenghts[track_idx])
+            else:
+                folder_name, chunk_start = self._get_track_from_chunk(index)
+            for s in self.sources:
+                if self.random_track_mix and self.random:
+                    track_idx = self._get_random_track_idx()
+                    folder_name, chunk_start = self.tracks[track_idx], self._get_random_start(
+                        self.track_lenghts[track_idx])
+                source_name = os.path.join(folder_name, s + '.wav')
+                audio, _ = torchaudio.load(
+                    source_name, frame_offset=chunk_start, num_frames=self.segment
+                )
+                if audio.shape[1] < self.segment:
+                    audio = F.pad(audio.unsqueeze(
+                        1), (0, self.segment - audio.shape[1])).squeeze(1)
+                stems.append(audio)
+            if self.random_track_mix and self.random:
+                x = sum(stems)
+            else:
+                x, _ = torchaudio.load(
+                    os.path.join(folder_name, 'mixture.wav'), frame_offset=chunk_start, num_frames=self.segment)
+                if x.shape[1] < self.segment:
+                    x = F.pad(x.unsqueeze(1),
+                              (0, self.segment - x.shape[1])).squeeze(1)
+
+        y = np.stack(stems)
```

Comment:
Sorry, I forgot to change this line to ```y = torch.stack(stems)```. You can fix it yourself or I can do it later.

Reply:
Don't worry, I can fix it.

Reply:
Actually, even without fixing it, `Dataloader` automatically converts them into torch tensors. Do you know if returning torch tensors inside `__getitem__` is faster than numpy arrays?

Reply:
Not sure though. The problem is that `torchaudio.load` is used so `stems` are all torch tensors.